subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2545,6 +2604,151 @@ void ull_conn_default_tx_time_set(uint16_t tx_time)
+				       &id, &ticks_current, &ticks_to_expire, &remainder,
+				       lazy, ticker_op_id_match_func, &ticker_id,
+				       ticker_get_offset_op_cb, (void *)&ret_cb);
+
+	if (ret == TICKER_STATUS_BUSY) {
+		while (ret_cb == TICKER_STATUS_BUSY) {
+			ticker_job_sched(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_LOW);
+		}
+	}
+
+	LL_ASSERT(ret_cb == TICKER_STATUS_SUCCESS);
+
+	/* Reduced a tick for negative remainder and return positive remainder
+	 * value.
+	 */
+	hal_ticker_remove_jitter(&ticks_to_expire, &remainder);
+	remainder_us = remainder;
+
+	/* Add a tick for negative remainder and return positive remainder
+	 * value.
+	 */
+	hal_ticker_add_jitter(&ticks_to_expire, &remainder);
+	start_us = remainder;
+
+	return HAL_TICKER_TICKS_TO_US(ticks_to_expire) + remainder_us - start_us;
+}
+
+static void mfy_past_sender_offset_get(void *param)
+{
+	uint16_t last_pa_event_counter;
+	uint32_t ticker_offset_us;
+	uint16_t pa_event_counter;
+	uint8_t adv_sync_handle;
+	uint16_t sync_handle;
+	struct ll_conn *conn;
+	uint16_t lazy;
+
+	conn = param;
+
+	/* Get handle to look for */
+	ull_lp_past_offset_get_calc_params(conn, &adv_sync_handle, &sync_handle);
+
+	if (adv_sync_handle == 0xFF && sync_handle == 0xFFFF) {
+		/* Procedure must have been aborted, do nothing */
+		return;
+	}
+
+	if (adv_sync_handle != 0xFF) {
+		struct ll_adv_sync_set *adv_sync = ull_adv_sync_get(adv_sync_handle);
+
+		LL_ASSERT(adv_sync);
+
+		ticker_offset_us = get_ticker_offset(TICKER_ID_ADV_SYNC_BASE + adv_sync_handle,
+						     &lazy);
+
+		pa_event_counter = adv_sync->lll.event_counter;
+		last_pa_event_counter = pa_event_counter - 1;
+	} else {
+		struct ll_sync_set *sync = ull_sync_is_enabled_get(sync_handle);
+		uint32_t interval_us = sync->interval * PERIODIC_INT_UNIT_US;
+		uint32_t window_widening_event_us;
+
+		LL_ASSERT(sync);
+
+		ticker_offset_us = get_ticker_offset(TICKER_ID_SCAN_SYNC_BASE + sync_handle,
+						     &lazy);
+
+		if (lazy && ticker_offset_us > interval_us) {
+
+			/* Figure out how many events we have actually skipped */
+			lazy = lazy - (ticker_offset_us / interval_us);
+
+			/* Correct offset to point to next event */
+			ticker_offset_us = ticker_offset_us % interval_us;
+		}
+
+		/* Calculate window widening for next event */
+		window_widening_event_us = sync->lll.window_widening_event_us +
+					   sync->lll.window_widening_periodic_us * (lazy + 1U);
+
+		/* Correct for window widening */
+		ticker_offset_us += window_widening_event_us;
+
+		pa_event_counter = sync->lll.event_counter + lazy;
+
+		last_pa_event_counter = pa_event_counter - 1 - lazy;
+
+		/* Handle unsuccessful events */
+		if (sync->timeout_expire) {
+			last_pa_event_counter -= sync->timeout_reload - sync->timeout_expire;
+		}
+	}
+
+	ull_lp_past_offset_calc_reply(conn, ticker_offset_us, pa_event_counter,
+				      last_pa_event_counter);
+}
+
+void ull_conn_past_sender_offset_get(struct ll_conn *conn)
```

Comment:
I understand how this naming can be confusing, however it does a `mayfly_enqueue`. When it runs it will send the result back using `ull_lp_past_offset_calc_reply`, which will set the result into the PDU and change the state in the llcp PAST state machine.

Would `ull_conn_past_sender_offset_request` be better?